xarray/core/dataarray.py

```diff
@@ -1369,6 +1369,61 @@ def real(self):
     @property
     def imag(self):
         return self._replace(self.variable.imag)
+    
+    def dot( self, other):
+        """Perform sum product of two DataArrays along their shared dims. 
+        Equivalent to taking taking tensor dot over all shared dims'''
+
+        Parameters
+        ----------
+        other: the DataArray with which the DataArray calling this function 
+        will perform dot with.
+
+        Returns
+        -------
+        DataArray(new_data, new_coords, new_dims) : DataArray
+            new_data: dot performed over shared DataArrays dims
+            new_dims: the dims that were not shared between DataArrays and 
+                      thus were not summed over.
+            new_coords: coords corresponding to those original unique dims.
+            
+        See also
+        --------
+        np.tensordot(a,b, axes)
+
+        Examples
+        --------
+        
+        >>> da.dims
+        ('x_trans', 'y_trans', 'imgID')  
+        >>> dm.dims
+        ('models', 'y_trans_m', 'imgID')
+   
+        >>> t = da.dot( dm )
+        >>> t.dims
+        ('x_trans', 'y_trans', 'models', 'y_trans_m')
+
+        """
+        s = self
+
+        if not ( isinstance(other, DataArray) ):
+            raise TypeError('dot only operates on DataArrays.')
+        
+        #sum over the common dims
+        dims = list(set(s.dims) & set(other.dims) )
+    
+        s, other = align(s, other, join='inner', copy=False)
+    
+        axes = (s.get_axis_num(dims), other.get_axis_num(dims))
+        f = ops._dask_or_eager_func('tensordot', n_array_args=2)
```

Comment:
I'm not as sure that I understand. Did you mean to move the bulk of the code to ops, and leave the dot operation in the DataArray class as largely just a thin wrapper around that operation.

Or what I initially thought you meant where when you said 'this' you were referring just to the last line in the code above. So I added this def to ops:
```
def tensordot(a, b, axes):
    #Perform dot product of two DataArrays along their shared dims.
    return _dask_or_eager_func('tensordot', n_array_args=2)(a,b,axes)
```
which just gets called on the last line in the referenced code above, to get the new data.